x/wasm-storage/keeper/abci.go

```diff
@@ -177,6 +154,62 @@ func (k Keeper) ExecuteTally(ctx sdk.Context) error {
 	return nil
 }
 
+func (k Keeper) filterAndTally(ctx sdk.Context, req Request) (vmRes tallyvm.VmResult, consensus bool, err error) {
+	filter, err := base64.StdEncoding.DecodeString(req.ConsensusFilter)
+	if err != nil {
+		return vmRes, consensus, fmt.Errorf("failed to decode consensus filter: %w", err)
+	}
+
+	// Sort reveals.
+	keys := make([]string, len(req.Reveals))
+	i := 0
+	for k := range req.Reveals {
+		keys[i] = k
+		i++
+	}
+	sort.Strings(keys)
+	reveals := make([]RevealBody, len(req.Reveals))
+	for i, k := range keys {
+		reveals[i] = req.Reveals[k]
+	}
+
+	outliers, consensus, err := ApplyFilter(filter, reveals)
+	if err != nil {
+		return vmRes, consensus, fmt.Errorf("error while applying filter: %w", err)
+	}
+
+	tallyID, err := hex.DecodeString(req.TallyBinaryID)
+	if err != nil {
+		return vmRes, consensus, fmt.Errorf("failed to decode tally ID to hex: %w", err)
+	}
+	tallyWasm, err := k.DataRequestWasm.Get(ctx, tallyID)
+	if err != nil {
+		return vmRes, consensus, fmt.Errorf("failed to get tally wasm for DR ID %s: %w", req.ID, err)
```

Comment:
Nitpicking but we should either add the DR ID to all errors or leave them out. Given that it's attached to the result for the DR ID I think we can leave it out :)